robosuite/utils/transform_utils.py

```diff
@@ -528,6 +552,45 @@ def rotation_matrix(angle, direction, point=None):
     return M
 
 
+def clip_rotation(quat, limit):
+    """
+    Limits a rotation to a specified limit
+
+    Converts rotation to axis-angle, clips, then re-converts back into quaternion
+
+    :param quat: Rotation being clipped (x, y, z, w) -- numpy array
+    :param limit: Value to limit rotation by -- magnitude (scalar)
+    :return: Clipped rotation quaternion (x, y, z, w)
+    """
+    if quat[3] == 1:
+        # This is a zero degree rotation, immediately return
+        return quat
+    elif quat[3] == 0:
```

Comment:
why is this if statement required? it seems like the case below would return the right values. is this due to rounding issues?